main/src/ca/uwaterloo/flix/language/phase/util/PhaseUtils.scala

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2025 Chenhao Gao
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ca.uwaterloo.flix.language.phase.util
+
+object PhaseUtils {
+  /**
+    * Merges two maps of lists by concatenating the lists with the same key.
+    */
+  def mergeMappedList[K, V](map1: Map[K, List[V]], map2: Map[K, List[V]]): Map[K, List[V]] = {
```

Comment:
I think we should add a companion object to ChangeSet and move this function there.
I would also rename it to mergeMapValues